### ISSUE TITLE
Allow unary no-plus-plus or no-minus-minus if operator is flush with the symbol (no whitespace)

### ISSUE THREAD

**Issue by marke-apexit** [eslint/eslint#18153]
### What rule do you want to change?

no-plus-plus & no-minus-minus

### What change do you want to make?

 Fail if there is an intervening whitespace between the ++ & -- unary operator and the symbol, as this is the real root of the issue that this rule is attempting to prevent.  However, also provides a way to allow the unary operator if it is flush with the symbol (no whitespace) as that is not susceptible to the semicolon insertion issue.

### How do you think the change should be implemented?

Not sure, a new option or separate rules?

### Example code

```js
// error or warning
foo ++
++ foo
foo --
-- foo

// allow
foo++
++foo
foo--
--foo
```

### What does the rule currently do for this code?

fails for all cases in the above code, with or without whitespace

### What will the rule do after it's changed?

Options could allow for disallowing any use, or only disallowing if the unary operator is not flush with the symbol.  Also if the flush unary operator is allowed, an autofix could be provided.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

I'm not sure the best way to implement this, options on the existing rule or separate rules, the team will no best.

**Comment by Tanujkanti4441** [comment]
Hi @marke-apexit, thanks for the issue, but is there any specific reason or issue you faced that led you to open this issue because in both the cases (with or without white space) the outputs are same i think.

**Comment by nzakas** [comment]
@Tanujkanti4441 don't forget to move the issue to "Triaging" in the Triage board when you start working on an issue.

@marke-apexit the `no-plusplus` rule simply bans the use of `++` or `--` irrespective of whitespace, as it's aim is to prevent using those operators completely. It would be inappropriate to add an opt-out for whitespace.
